### ISSUE TITLE
email_text_input: 'initial value must be a valid email format' error when specifying multiple email addresses

### ISSUE THREAD

**Issue by snobear** [slackapi/node-slack-sdk#1580]
Hi, 
I'm trying to specify multiple emails in an `email_text_input` field. 

It's not documented, but in general it works if you specify emails, space separated. The input passes validation in the view modal and then submits the space-separated email string in the payload OK.

However, when trying to specify multiple space-separated emails as the `initial_value`, I get:

```
[ERROR]  web-api:WebClient:1 initial value must be a valid email format [json-pointer:view/blocks/1/element/initial_value]
```

#### Packages:

Select all that apply:

- [x] `@slack/web-api`
- [ ] `@slack/rtm-api`
- [ ] `@slack/webhooks`
- [ ] `@slack/oauth`
- [ ] `@slack/socket-mode`
- [ ] `@slack/types`
- [ ] I don't know

#### The Slack SDK version

```bash
"slack/web-api": "^6.7.2",
```

#### Node.js runtime version

```bash
node --version
v17.7.1
```

#### OS info

```bash
sw_vers && uname -v
ProductName:    macOS
ProductVersion: 11.6.5
BuildVersion:   20G527
Darwin Kernel Version 20.6.0: Tue Feb 22 21:10:41 PST 2022; root:xnu-7195.141.26~1/RELEASE_X86_64
```

#### Steps to reproduce:

An example view call looks like:

```javascript
const view = {
  type: 'modal',
  callback_id: 'foo',
  title: {
    type: 'plain_text',
    text: 'Reply All',
  },
  submit: {
    type: 'plain_text',
    text: 'Reply'
  },
  blocks: [
    {
      type: 'input',
      element: {
        type: 'email_text_input',
        action_id: 'to',
        initial_value: 'jase@example.com',
      },
      label: {
        type: 'plain_text',
        text: 'To:'
      }
    },
    {
      type: 'input',
      element: {
        type: 'email_text_input',
        action_id: 'cc',
        initial_value: 'abc@example.com bahh@example.com dkdk@example.com',
      },
      label: {
        type: 'plain_text',
        text: 'CC:'
      }
    },
    {
      type: 'input',
      element: {
        type: 'plain_text_input',
        action_id: 'reply_body',
        multiline: true,
        focus_on_load: true,
      },
      label: {
        type: 'plain_text',
        text: 'Body'
      }
    }
  ],
}

await slack.views.open({
  channel: 'channelidhere',
  trigger_id: 'sometriggerid',
  view,
});
```

### Expected result:

I can specify multiple space-separated emails as `inital_value` for `email_text_input` without an error.

### Actual result:

When calling `views.open`:
```
[ERROR]  web-api:WebClient:1 initial value must be a valid email format [json-pointer:view/blocks/1/element/initial_value]
```

### Requirements

For general questions/issues about Slack API platform or its server-side, could you submit questions at https://my.slack.com/help/requests/new instead. :bow:

Please read the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

**Comment by srajiang** [comment]
Hey @snobear - I can see how it would be surprising that if email_text_input accepts this type of string with multiple emails as an input but doesn't accept the same value for `initial_value`. 

I don't think this behavior is a bug though - it seems the api is letting app developers handle validation on user-inputs in their own desired way, but ensures at a minimum that when a user is shown a view with an initial_value, that what the app developer has supplied as that value conforms to a standard email address format.

**Comment by snobear** [comment]
Hi @srajiang , thanks for the reply!

IMO the UI should also fail validation if only single email addresses are supported. i.e., it should fail if spaces are detected (like when attempting to enter multiple space-separated emails), since [spaces are not allowed](https://www.rfc-editor.org/rfc/rfc2822#section-3.4.1) in email addresses.

That said, can I make a feature request to allow `email_text_input` to accept one or more emails? Let me know if I should open a new issue or if can just relabel this one.

**Comment by srajiang** [comment]
@snobear - Fair point, I do agree that a better design would be to figure out how to properly allow multiple email addresses to be input. 

Let me submit this as a a request with the team that works on Block Kit and keep this issue open while it's outstanding. 

In the meantime, another way to surface product requests is directly through https://my.slack.com/help/requests/new.

**Comment by srajiang** [comment]
@snobear - Letting you know that the server teams have looked into this and it seems as though letting users submit a string in this type of format 'abc@example.com bahh@example.com dkdk@example.com' is soon going to be disallowed, as the email text input is intended to be a single input field. I'm sure that's probably not the news you wanted to hear. 

They are going to explore officially supporting multiple email addresses in CSV format, which is the more correct longterm solution, but unfortunately I don't have a date for when we can expect that. That being said, the request is definitely on th teams radar. 

For your current use-case of wanting to accept input like the above, I would recommend going with a `plain text input` for now. There shouldn't be validation on `initial_value` on that field, so you can supply an initial value in the format you described above. 

You might explore adding a context element with instructions on formatting the list of email addresses for your users.

**Comment by snobear** [comment]
Great, thanks for submitting that. 

A plain text input field will work fine for me. Having the built in validation on the front end side would be great, but not a blocker. 

Csv makes more sense going forward, and totally understand it'd be a separate initiative to tackle.

I think we can close this issue. Thanks again! It was good just to get some clarity on it.

**Comment by srajiang** [comment]
Glad that you can work around this for now! I'll close this up now.
